perun-base/src/main/java/cz/metacentrum/perun/core/api/AttributeDefinition.java

```diff
@@ -134,6 +134,14 @@ public void setUnique(boolean unique) {
 		this.unique = unique;
 	}
 
+	/**
+	 * Returns true if the attribute definition is for a virtual attribute.
+	 *
+	 * @return true if the attribute definition is for a virtual attribute, false otherwise.
+	 */
+	public boolean isVirtual() {
+		return this.namespace.endsWith("virt");
+	}
```

Comment:
Did you add this to serialization and de-serialization? We probably can skip it.

Reply:
I have edited the serializers to skip the value.

Reply:
While this is technically/java correct solution, I would rather avoid adding such specific method in an object, exactly because it has to be handled in (de)serializer. Method returning type of attribute definition would be more general solution (like we do in ``getEntity()``).
After this change, it has only single usage. We have check methods in ``AttributesManagerImpl`` or some utils class. They are not generic, so we can create a new one or simply check "virtual" state in that single usage place.

PS: If you still want to keep your solution, then please update also (de)serializer in rpclib package. Otherwise we won't be able to read back the AttributeDefinition object from auditer log.

Reply:
I have changed the check that it is using the method from impl layer.